### ISSUE TITLE
Run compilestatic without a database

### ISSUE THREAD

**Issue by wesleykendall** [andreyfedoseev/django-static-precompiler#113]
Is it possible to run `compilestatic` without needing a database? Specifically for less compilation.

**Comment by andreyfedoseev** [comment]
The database is used to store a list of dependencies for the compiled files. This is necessary to recompile the main file when one of the imported files were changed. If you don't need dependency tracking then you can create your custom LESS compiler like so:

```
class LESS(static_precompiler.compilers.less.LESS):
     supports_dependencies = False
```

and register it in your project's settings:

```
STATIC_PRECOMPILER_COMPILERS = (
    'custom_compiler.LESS',
)
 ```

There's one problem with that, though. I just looked through the code and found one place where the `supports_dependencies` attribute is not taken into account: https://github.com/andreyfedoseev/django-static-precompiler/blob/master/static_precompiler/compilers/base.py#L313

As a workaround for that you can override `get_dependents` method in your custom compiler to make it return an empty list.

**Comment by wesleykendall** [comment]
I was literally just about to respond that I solved this with:

```
class LESSNoDB(less_compiler.LESS):
    """
    The LESS compiler from django-static-precompiler
    with the database turned off.

    Turning off dependencies prevents rebuilds from being triggered
    from dependencies being changed. This is not needed
    in production compilations
    """
    supports_dependencies = False

    def get_dependencies(self, *args, **kwargs):
        return []

    def get_dependents(self, *args, **kwargs):
        return []

    def update_dependencies(self, *args, **kwargs):
        return []
```

Good to know that I only need to override `get_dependents`. Thanks for the speedy response!

Out of curiosity, does this turn off any features? Like automatic compilation if a dependency changes? Feel free to close this if you'd like :)

**Comment by mrmachine** [comment]
Can this be reopened? Not having a way to `manage.py compilestatic` while "offline" makes this impossile to use with Docker when scaling across multiple servers. E.g. when building a `Dockerfile` into an image, the image needs to include all precompiled assets so all servers are able to satisfy those requests.

At the very least, the `LESSNoDB` workaround above should be bundled as an option or documented.

**Comment by andreyfedoseev** [comment]
@mrmachine 

I just added a new option to the `compilestatic` command (`--ignore-dependencies`) that disables dependency tracking and prevents database access. It isn't uploaded to PyPI yet (although, I plan the update to be released soon). If you have time, could you test it using the latest `master` branch and let me know if it works for your case?

**Comment by andreyfedoseev** [comment]
I just released version 1.8 with `--ignore-dependencies` option added.
